sql/core/src/main/scala/org/apache/spark/sql/execution/command/ddl.scala

```diff
@@ -274,6 +274,80 @@ case class AlterTableUnsetPropertiesCommand(
 
 }
 
+
+/**
+ * A command to change the columns for a table, only support changing the comments of non-partition
+ * columns for now.
+ *
+ * The syntax of using this command in SQL is:
+ * {{{
+ *   ALTER TABLE table_identifier
+ *   CHANGE [COLUMN] column_old_name column_new_name column_dataType [COMMENT column_comment]
+ *   [FIRST | AFTER column_name];
+ * }}}
+ */
+case class AlterTableChangeColumnsCommand(
+    tableName: TableIdentifier,
+    columns: Map[String, StructField]) extends RunnableCommand {
+
+  // TODO: support change column name/dataType/metadata/position.
+  override def run(sparkSession: SparkSession): Seq[Row] = {
+    val catalog = sparkSession.sessionState.catalog
+    val table = catalog.getTableMetadata(tableName)
+    val resolver = sparkSession.sessionState.conf.resolver
+    DDLUtils.verifyAlterTableType(catalog, table, isView = false)
+
+    // Create a map that converts the origin column to the new column with changed comment, throw
+    // a Exception if the column reference is invalid or the column name/dataType is changed.
+    val columnsMap = columns.map { case (oldName: String, newField: StructField) =>
+      // Find the origin column from schema by column name.
+      val originColumn = findColumn(table.schema, oldName, resolver)
+      // Throw a Exception if the column name/dataType is changed.
+      if (!columnEqual(originColumn, newField, resolver)) {
+        throw new AnalysisException(
+          "ALTER TABLE CHANGE COLUMN is not supported for changing column " +
+            s"'${getDesc(originColumn)}' to '${getDesc(newField)}'")
+      }
+      // Create a new column from the origin column with new comment.
+      val newColumn = addComment(originColumn, newField.getComment)
+      // Create the map from origin column to changed column
+      originColumn -> newColumn
+    }
+
+    val newSchema = table.schema.fields.map(field => columnsMap.getOrElse(field, field))
+    val newTable = table.copy(schema = StructType(newSchema))
+    catalog.alterTable(newTable)
+
+    Seq.empty[Row]
+  }
+
+  // Find the origin column from schema by column name, throw a Exception if the column
+  // reference is invalid.
+  private def findColumn(schema: StructType, name: String, resolver: Resolver): StructField = {
+    schema.fields.collectFirst {
+      case field if resolver(field.name, name) => field
+    }.getOrElse(throw new AnalysisException(
+      s"Invalid column reference '$name', table schema is '${schema}'"))
+  }
+
+  // Add the comment to a column, if comment is empty, return the original column.
```

Comment:
Hive won't change the original comment if the COMMENT statement is empty. We keep the same with this behavior.